### PR TITLE
Add Depth Anything V3 to model zoo

### PR DESCRIPTION
**Pull request by CharlesCNorton**
## Summary

- Add Depth Anything V3 models to the model zoo
- Create `fiftyone/utils/depth_anything.py` wrapper

## Models Added

| Model | Size | Description |
|-------|------|-------------|
| `depth-anything-v3-small-torch` | 99 MB | Lightweight, fast inference |
| `depth-anything-v3-base-torch` | 541 MB | Balanced performance |
| `depth-anything-v3-large-torch` | 1.64 GB | High accuracy |

## Usage

```python
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart", max_samples=5)

model = foz.load_zoo_model("depth-anything-v3-base-torch")
dataset.apply_model(model, label_field="depth")

session = fo.launch_app(dataset)
```

## Test plan

- [x] Model loading and inference
- [x] Output normalized heatmaps
- [x] Multiple input types (file path, PIL, NumPy, Tensor)

## Summary by CodeRabbit

* **New Features**
  * Added Depth Anything V3 integration with on-demand installation, image/batch support, per-image normalization and resizing, optional confidence/sky outputs, multi-view depth computation, and 3D export helpers (GLB/PLY).
* **Tests**
  * Added unit tests covering configuration, output processing, normalization, resizing, dtype handling, batching, zero-depth and other edge cases.

### CODE REVIEW

File: fiftyone/utils/depth_anything.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+import torch
+
+logger = logging.getLogger(__name__)
+
+
+def _ensure_depth_anything_3():
+    if not fou.ensure_package("depth-anything-3", error_level=2):
+        fou.install_package(
+            "git+https://github.com/ByteDance-Seed/depth-anything-3.git"
+            "@2c21ea849ceec7b469a3e62ea0c0e270afc3281a"
```
Comment (lorenzoriano) [inline fiftyone/utils/depth_anything.py:33]: Is this right? Pinning the zoo model to a specific commit?

File: fiftyone/utils/depth_anything.py
Comment (lorenzoriano) [inline fiftyone/utils/depth_anything.py:1]: can you add type hints to the functions?

File: fiftyone/utils/depth_anything.py [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+                conf_maps = conf_maps[np.newaxis, ...]
+
+        sky_masks = output.get("sky")
+        if sky_masks is not None:
+            if isinstance(sky_masks, torch.Tensor):
+                sky_masks = sky_masks.detach().cpu().numpy()
+            if len(sky_masks.shape) == 2:
+                sky_masks = sky_masks[np.newaxis, ...]
+
+        from PIL import Image
```
Comment (lorenzoriano) [inline fiftyone/utils/depth_anything.py]: why the import here instead of top of the file?

File: fiftyone/utils/depth_anything.py [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+
+        width, height = frame_size
+        results = []
+
+        for i, depth in enumerate(depth_maps):
```
Comment (lorenzoriano) [inline fiftyone/utils/depth_anything.py:143]: somewhere you should have a check for None or empty depth_maps

File: fiftyone/utils/depth_anything.py [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+        dataset.apply_model(model, label_field="depth")
+
+        session = fo.launch_app(dataset)
+
+    Args:
+        config: a :class:`DepthAnythingV3ModelConfig`
+    """
+
+    def _download_model(self, config):
+        pass
```
Comment (lorenzoriano) [inline fiftyone/utils/depth_anything.py]: I think the model downloading part should be here, and `_load_model` will just load it

File: fiftyone/utils/depth_anything.py [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+                logger.warning(
+                    "infer_gs=True but model returned no gaussians. "
+                    "Only giant and nested models support Gaussian Splatting"
+                )
+        if prediction.aux:
+            output["aux"] = prediction.aux
+        return output
+
+
+def compute_multiview_depth(
```
Comment (lorenzoriano) [inline fiftyone/utils/depth_anything.py]: I'm not sure it's good to have a standalone function. Is that common in the repo? If I call it several times it'll create the model over and over. What about having it as a method of `DepthAnythingV3Model`?

```
    def compute_multiview_depth(self, filepaths, **kwargs):
        prediction = self._model.inference(
            filepaths,
            process_res=self.config.process_res,
            use_ray_pose=kwargs.get('use_ray_pose', False),
            ...
        )
```

File: fiftyone/utils/depth_anything.py (hunk 2 above)
Comment (CharlesCNorton) [reply fiftyone/utils/depth_anything.py]: no pypi package or release tags so commit pin is their latest. Will unpin when they publish.

File: fiftyone/utils/depth_anything.py
Comment (lorenzoriano) [reply fiftyone/utils/depth_anything.py:1]: There are still functions that don't have type hints, can you (auto-) add them?
